fix date start and date end in massaction

ATM-NicolasV opened the pull request:
# FIX|Fix #[*issue_number Short description*]
Bugfix: Robust Date Comparison for MO (Manufacturing Order) Planned Dates

Problem

The existing validation logic for modifying the planned start and end dates of a Manufacturing Order (MO) fails when one of the opposing dates is not yet defined (stored as an empty string "").

This issue is caused by PHP's loose comparison :

The positive timestamp $newDate is compared against the empty string "".

PHP implicitly converts "" to the integer 0.

The comparison (e.g., timestamp < 0) evaluates to FALSE, prematurely executing the error block (setEventMessages) and preventing the user from successfully setting the initial date.

Solution

The fix standardizes the object's date attributes ($objMo->date_end_planned, etc.) into reliable integer timestamps before performing validation.

We use is_numeric() to detect existing timestamps (positive integers).

We use strtotime() to safely convert empty strings ("" becomes 0) or date strings to integers.

The validation condition is updated to explicitly allow the modification if the opposing date is undefined (value $\le 0$ after conversion).

ATM-NicolasV committed:
fix date start and date end in massaction

ATM-NicolasV committed:
PR returns : change comment langs and restore comment mandatroty for travis

ATM-NicolasV reviewed (commented):
I have make change for comments

eldy commented on htdocs/mrp/mo_list.php:
```diff
@@ -253,34 +253,32 @@
-										break;
-									}
+						if (!empty($changeDate)) {
+							if ($action == 'changedatestart_confirm') { 		// Test on permission
```
Must be   // Test on permission not required

ATM-NicolasV committed:
Clarify permission check for date change action

Updated comment to clarify that permission check is not required for date change action.
